### DISCUSSION TITLE
How does hosted Supabase selects the realtime server (Stream vs RLS)?

### DISCUSSION THREAD

**Discussion by osaxma** [supabase/supabase#9498] [Questions] [answered]
I'm just curious how the hosted Supabase chooses which realtime server version to run?

In the [realtime readme](https://github.com/supabase/realtime), the following is outlined:
> There are two versions of this server: `Realtime` and `Realtime RLS`.
> 
> `Realtime` server works by:  [<~~ aka "Stream"]
> 
> 1. listening to PostgreSQL's replication functionality (using PostgreSQL's logical decoding)
> 2. converting the byte stream into JSON
> 3. broadcasting to all connected clients over WebSockets
> 
> `Realtime RLS` server works by:
> 
> 1. polling PostgreSQL's replication functionality (using PostgreSQL's logical decoding and wal2json output plugin)
> 2. passing database changes to a Write Ahead Log Realtime Unified Security (WALRUS) PostgresSQL function and receiving a list of authorized subscribers depending on Row Level Security (RLS) policies
> 3. converting the changes into JSON
> 3. broadcasting to authorized subscribers over WebSockets

As I was rereading the Realtime RLS [announcement](https://supabase.com/blog/realtime-row-level-security-in-postgresql), I came across the following:

> Disable for public tables
> If your data is insensitive or publicly available, such as stock prices listed under NASDAQ, then don't enable row level security!
> 
> The fastest security policy is one that doesn't exist :)

Given that, I assume running either Realtime (Stream) or Realtime RLS is optional but I'm uncertain how that option can be selected.

Does this mean that if any table has RLS enabled, then hosted Supabase will use Realtime RLS? Or does hosted Supabase uses Realtime RLS by default nevertheless? Or does Supabase run two servers one for RLS and one for Stream?

**Comment by w3b6x9** [comment, marked as answer]
@osaxma hosted Supabase runs Realtime RLS for all projects at the moment. In the near future, you'll be able to choose whether you want the streaming or RLS version.
